### PR TITLE
fix: assume unknown tx type is `TransactionType.Legacy`

### LINKED ISSUE #3458: Forking Arbitrum Nitro chains gives "Invalid transaction type"
Running `ganache --fork <url>` with an Arbitrum Nitro network returns `Error: Invalid transaction type: 106`.  I have attempted with Alchemy and QuickNode endpoints, and tried both Arbitrum Rinkeby and Arbitrum Goerli.

```console
ganache v7.4.0 (@ganache/cli: 0.5.0, @ganache/core: 0.5.0)
Starting RPC server
Error: Invalid transaction type: 106
    at Function.typeOf ([myproject]/node_modules/ganache/dist/node/1.js:2:218592)
    at Function.typeOfRPC ([myproject]/node_modules/ganache/dist/node/1.js:2:218818)
    at Function.fromRpc ([myproject]/node_modules/ganache/dist/node/1.js:2:217517)
    at [myproject]/node_modules/ganache/dist/node/1.js:2:90737
    at Array.forEach (<anonymous>)
    at Function.rawFromJSON ([myproject]/node_modules/ganache/dist/node/1.js:2:90566)
    at Fork.initialize ([myproject]/node_modules/ganache/dist/node/1.js:2:108321)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Blockchain.initialize ([myproject]/node_modules/ganache/dist/node/1.js:2:74897)
    at async EthereumProvider.initialize ([myproject]/node_modules/ganache/dist/node/1.js:2:161074)
```

### LINKED ISSUE #3489: Transactions with a `type` field Ganache doesn't understand should be treated as "untyped"
Related: https://github.com/trufflesuite/ganache/issues/3458

### PR DESCRIPTION
**Pull request by jaketimothy**
Fixes #3489
Fixes #3458

### CODE REVIEW

**Comment by jaketimothy** [comment]
Let me know if this is too permissive

**Review by MicaiahReid — APPROVED** [review]
I think I'm cool with this. Thanks for the PR!

I would like to note - this should fix your problems with Arbitrum Nitro throwing with an "invalid transaction types" error, but we still won't actually support these transaction types. Or any non EVM compatible chains. (I know, Arbitrum claims to be EVM compatible, then they go and start making up transaction types! :sweat_smile: )

We are working on a plugin system so that users can extend Ganache to work with different L2s. Keep an eye on our releases for when this will be available. Thanks again!

File: src/chains/ethereum/transaction/tests/index.test.ts [hunk 1]
```diff
@@ -144,6 +150,12 @@ describe("@ganache/ethereum-transaction", async () => {
         );
         assert.strictEqual(txFromRpc.type.toString(), "0x0");
       });
+      it("assumes legacy transaction if type unknown", () => {
+        txFromRpc = <LegacyTransaction>(
+          TransactionFactory.fromRpc(unknownTypeTx, common)
+        );
+        assert.strictEqual(txFromRpc.type.toString(), "0x0");
```
Comment (davidmurdoch) [inline src/chains/ethereum/transaction/tests/index.test.ts:183]: Can you also add a test to handle the case where the unknown `type` is `"0xc0"` and one more where it some value greater than `"0xc0"`?

**Review by davidmurdoch — CHANGES_REQUESTED** [review]
Awesome! Thanks for doing the work for this. I just have one change request.

**Comment by davidmurdoch** [comment]
Thanks @jaketimothy for those quick test updates. Tests are running now and we'll get this approved and merged soon!

**Comment by jaketimothy** [comment]
@davidmurdoch lmk if there's anything more I can take care of here!

**Review by davidmurdoch — APPROVED** [review]
Looks perfect! Thanks!

**Pull request merged**